File: python/double_pointers/Solution11.py
# 11. 盛最多水的容器
from typing import List


class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        left = 0
        right = len(height) - 1
        while left < right:
            area = (right - left) * min(height[right], height[left])
            res = max(area, res)
            if height[right] >= height[left]:
                left += 1
            else:
                right -= 1
        return res

File: src/doublepointers/Solution11.java
package doublepointers;

// 11. 盛最多水的容器
public class Solution11 {

    public int maxArea(int[] height) {

        int max = 0;
        int left = 0;
        int right = height.length - 1;
        while (left < right) {
            int area = (right - left) * Math.min(height[right], height[left]);
            max = Math.max(area, max);
            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }
        return max;

    }
}
